src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -313,12 +313,18 @@ object ConnectedComponents extends Logging {
 
     var converged = false
     var iteration = 1
-    var prevSum: BigDecimal = null
+
+    // compute min neighbors (including self-min)
+    var minNbrs1: DataFrame = minNbrs(ee) // src >= min_nbr
+      .persist(intermediateStorageLevel)
+
+    var prevSum: BigDecimal = minNbrs1.select(sum(col(MIN_NBR).cast(DecimalType(20, 0))), count("*")).rdd
```

Comment:
Is there any concern with an overflow by capping this at 20?

Reply:
added verification.